test/unit/test_selector_errors.py

```diff
@@ -0,0 +1,71 @@
+from dbt.graph import cli
+import dbt.exceptions
+import textwrap
+import yaml
+import unittest
+from pprint import pprint
+from dbt.config.selectors import (
+    selector_config_from_data
+)
+
+from dbt.contracts.selection import SelectorFile
+
+
+def get_selector_dict(txt: str) -> dict:
+    txt = textwrap.dedent(txt)
+    dct = yaml.safe_load(txt)
+    return dct
+
+
+
+class SelectorUnitTest(unittest.TestCase):
+
+    def test_parse_multiple_excludes(self):
+        dct = get_selector_dict('''\
+            selectors:
+                - name: mult_excl
+                  definition:
+                    union:
+                      - method: tag
+                        value: nightly
+                      - exclude:
+                         - method: tag
+                           value: hourly
+                      - exclude:
+                         - method: tag
+                           value: daily
+            ''')
+        with self.assertRaises(dbt.exceptions.DbtSelectorsError):
+            selectors = selector_config_from_data(dct)
+
+    def test_parse_set_op_plus(self):
+        dct = get_selector_dict('''\
+            selectors:
+                - name: union_plus
+                  definition:
+                    - union:
+                       - method: tag
+                         value: nightly
+                       - exclude:
+                          - method: tag
+                            value: hourly
+                    - method: tag
+                      value: foo
```

Comment:
Could you also add an exception + test for quite similar case, which I've seen a few folks run into:
```yml
selectors:
  - name: mixed_syntaxes
    definition:
      key: value
      method: tag
      value: foo
      union:
        - method: tag
          value: nightly
        - exclude:
          - method: tag
            value: hourly
```
The issue here is that contradictory dict arguments are being passed to `definition`, and it's not at all clear which ones it's using.

Reply:
Are we talking about the "key: value" piece? or having a union in a dictionary without a 'union' above it? I was under the impression that the above should be:

`selectors:
    - name: mixed_syntaxes
      definition:
         union:
           - method: tag
              value: foo
            - method: tag
               value: nightly
            - exclude:
                 method: tag
                 value: hourly
`
So having a 'union' in a single dictionary should be an error, and having an invalid keyword 'key' should be an error?

Reply:
Right on—the way you've defined it is how it _should_ be. I've seen users write the version I put above, which neither throws an error nor works the way they'd expect. I think we should check to see that `definition` is one of:
- string
- dict w/ 1 argument (`union`, `intersection`, or arbitrary `key: value`)
- dict w/ 2 arguments (_must_ be `method` + `value`)

Reply:
I think there are quite a few other arguments that you *can* put in a dict, such as 'children', 'parents', etc. What I did and pushed was to check that if there is a union or intersection operator in the root level, there's not also a 'method' key. The way the code worked was that it would just. ignore additional dict keys if there was a union or intersection, so that takes the "way it works" and creates an error. 

Let me know if you see any other situations that are problematic.

Reply:
Ah, that's a really good point about `children`, `parents`, `child_depth`, etc. Silly of me to forget, sorry.

That sounds about right to me. Would it be possible to generalize this:

> check that if there is a union or intersection operator in the root level, there's not also a 'method' key.

To be: "if there is a union or intersection operator in the root level, there should be no other dict arguments."

I think this would also avoid the weirdness of something like:
```yml
selectors:
    - name: multiple_sets
      definition:
         intersection:
           - method: tag
             value: foo
           - method: tag
             value: bar
         union:
           - method: tag
             value: nightly
           - exclude:
              - method: tag
                value: hourly
```

If I understand correctly, the current behavior would be to take the first set and ignore the second set entirely. Really we should throw an error.

Reply:
Yes, that would cover more situations. Will change.

Reply:
It looks like this now raises a much better error message:
```yml
selectors:
    - name: intersection_and_method
      definition:
         intersection:
           - method: tag
             value: foo
           - method: tag
             value: bar
         method: tag
         value: bar
```
```
 You cannot have both method and intersection keys in a root level selector definition:
```
Nice!

The `multiple_sets` definition above, however, still doesn't raise an error.

Reply:
Strange.  For the multiple_set example I get: Only a single 'union' or 'intersection' key is allowed in a root level selector definition; found intersection,union.

Reply:
This now works wonderfully!
```
Could not read selector file data: Runtime Error
    Only a single 'union' or 'intersection' key is allowed in a root level selector definition; found intersection,method,value.
```
I'm sure this was on me for not pulling the latest remote changes.